test/e2e/agent/agent_test.go

```diff
@@ -63,4 +64,317 @@ var _ = Describe("VM Agent behavior", func() {
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+		})
+
+		It("should report the correct device status during and after a device os image update", func() {
+			// Should report 'Online', 'Reboot', 'Online' after a device os image update.
+			deviceId := harness.GetEnrollmentIDFromConsole()
+
+			// Approve the enrollment and wait for the device details to be populated by the agent.
+			harness.ApproveEnrollment(deviceId, testutil.TestEnrollmentApproval())
+			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
+				deviceId).ShouldNot(BeEmpty())
+			logrus.Infof("The device %s was approved", deviceId)
+
+			// Wait for the device to pickup enrollment and report measurements on device status.
+			Eventually(harness.GetDeviceWithStatusSystem, TIMEOUT, POLLING).WithArguments(
+				deviceId).ShouldNot(BeNil())
+			logrus.Infof("The device %s is reporting its status", deviceId)
+
+			// Check the device status right after bootstrap.
+			response := harness.GetDeviceWithStatusSystem(deviceId)
+			device := response.JSON200
+			Expect(device.Status.Summary.Status).To(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+			Expect(*device.Status.Summary.Info).To(Equal("Bootstrap complete"))
+			Expect(device.Status.Updated.Status).To(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
+
+			var newImageReference string
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				currentImage := device.Status.Os.Image
+				logrus.Infof("Current image for %s is %s", deviceId, currentImage)
+				repo, _ := parseImageReference(currentImage)
+				newImageReference = repo + ":v2"
+				device.Spec.Os = &v1alpha1.DeviceOSSpec{Image: newImageReference}
+				logrus.Infof("Updating %s to image %s", deviceId, device.Spec.Os.Image)
+			})
+
+			harness.WaitForDeviceContents(deviceId, "the device is upgrading to renderedVersion: 2",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "True", "Update")
+				}, "1m")
+			Expect(device.Status.Summary.Status).To(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+
+			harness.WaitForDeviceContents(deviceId, "the device is rebooting",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "True", "Rebooting")
+				}, "2m")
+			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Rebooting")))
+
+			harness.WaitForDeviceContents(deviceId, "status.Os.Image gets updated",
+				func(device *v1alpha1.Device) bool {
+					return device.Status.Os.Image == newImageReference &&
+						conditionExists(device, "Updating", "False", "Updated")
+				}, "2m")
+			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
```

Comment:
Aren't we repeating: https://github.com/flightctl/flightctl/blob/main/test/e2e/agent/agent_updates_test.go#L32 ? 

Should we just incorporate any changes to the other test instead?

(given that it's expensive to run)